### PR TITLE
Point assignment operator

### PR DESCRIPTION
**Pull request by dangars**
Following up on the comment https://github.com/dealii/dealii/pull/8384#pullrequestreview-265055555

I added a point assignment operator between two points with different underlying type.

### CODE REVIEW

File: include/deal.II/base/point.h [hunk 1]
```diff
@@ -195,6 +195,17 @@ class Point : public Tensor<1, dim, Number>
...
 
+  /**
+   * Assignment operator from Tensor<1, dim, Number> with different underlying
+   * scalar type. This obviously requires that the @p OtherNumber type is
+   * convertible to @p Number.
+   *
+   * @note This function can also be used in CUDA device code.
+   */
+  template <typename OtherNumber>
+  DEAL_II_CUDA_HOST_DEV Point<dim, Number> &
```
Comment (masterleinad) [inline include/deal.II/base/point.h]:
```suggestion
Point<dim, Number> &
```
Let's be cautious here. Assigning between different Number types in `CUDA` code doesn't turn out to work well and leads to hard to understand errors.

**Comment by masterleinad** [comment]
Thanks! You also need to run the indentation script.

**Comment by dangars** [comment]
Now I think it should be ok :)

**Comment by masterleinad** [comment]
Thanks!

File: include/deal.II/base/point.h [hunk 2]
```diff
@@ -195,6 +195,17 @@ class Point : public Tensor<1, dim, Number>
   DEAL_II_CUDA_HOST_DEV Number &
                         operator()(const unsigned int index);
 
+  /**
+   * Assignment operator from Tensor<1, dim, Number> with different underlying
+   * scalar type. This obviously requires that the @p OtherNumber type is
+   * convertible to @p Number.
+   *
```
Comment (masterleinad) [inline include/deal.II/base/point.h]:
```suggestion
```

File: include/deal.II/base/point.h [hunk 3]
```diff
@@ -195,6 +195,17 @@ class Point : public Tensor<1, dim, Number>
...
+   * @note This function can also be used in CUDA device code.
```
Comment (masterleinad) [inline include/deal.II/base/point.h]: Ah, I missed that there is still this comment. Would you removing it?
```suggestion
```

Comment (dangars) [reply include/deal.II/base/point.h]: Sorry, I forgot to remove that comment. Now it is done!

**Pull request merged**